clients/input_copy.go

```diff
@@ -32,9 +36,24 @@ type InputCopy struct {
 	SourceOutputUrl string
 }
 
+type DecryptionKeys struct {
+	decryptKey   rsa.PrivateKey
+	encryptedKey string
+}
+
 // CopyInputToS3 copies the input video to our S3 transfer bucket and probes the file.
-func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL) (inputVideoProbe video.InputVideo, signedURL string, osTransferURL *url.URL, err error) {
-	if isDirectUpload(inputFile) {
+func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL, encryptedKey string, VodDecryptPrivateKey *rsa.PrivateKey) (inputVideoProbe video.InputVideo, signedURL string, osTransferURL *url.URL, err error) {
```

Comment:
What if we just got the `*DecryptionKeys` arg here? Will probably be cleaner overall since there's already a conditional on the `coordinator` today